source/base/DateTime.ooc

```diff
@@ -14,22 +14,251 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-TimeSpan: cover {
-	_ticks: Int64
-	ticks ::= this _ticks
-	init: func@ (=_ticks)
-	kean_base_timeSpan_getTicks: unmangled func -> Int64 { this _ticks }
-	new: unmangled(kean_base_timeSpan_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
+
+use ooc-base
+import os/Time
+
+DateTimeData: class {
+	year: Int { get set }
+	month: Int { get set }
+	day: Int { get set }
+	hour: Int { get set }
+	minute: Int { get set }
+	second: Int { get set }
+	millisecond: Int { get set }
+	init: func@ (year, month, day, hour, minute, second, millisecond: Int) {
+		this year = year
+		this month = month
+		this day = day
+		this hour = hour
+		this minute = minute
+		this second = second
+		this millisecond = millisecond
```

Comment:
I'm not 100% sure this is going to work with properties, but, it might be possible to write this as 
```ooc
DateTimeData: class {
	year: Int { get set }
	month: Int { get set }
	day: Int { get set }
	hour: Int { get set }
	minute: Int { get set }
	second: Int { get set }
	millisecond: Int { get set }
	init: func@ (=year, =month, =day, =hour, =minute, =second, =millisecond)
}
```
Give it a try!